proto/spacemesh/v1/types.proto

```diff
@@ -79,18 +44,29 @@ message Activation {
 // An immutable Spacemesh transaction.
 // do not include mutable data such as tx state or result.
 message Transaction {
-    TransactionId id = 1;
-    oneof datum {
-        CoinTransferTransaction coin_transfer = 2;
-        SmartContractTransaction smart_contract = 3;
-    }
-    AccountId sender = 4; // tx originator, should match signer inside Signature
-    GasOffered gas_offered = 5; // gas price and max gas offered
-    Amount amount = 6; // amount of coin transfered in this tx by sender
-    uint64 counter = 7; // tx counter aka nonce
-    Signature signature = 8; // sender signature on transaction
+    bytes id = 1;
+    bytes principal = 2;
+    bytes template = 3;
+    uint32 method = 4;
```

Comment:
In our design this is actually `uint8` as well, right?

Reply:
yes